crypto/evp_extra/p_dh.c

```diff
@@ -121,6 +122,29 @@ static int pkey_dh_ctrl(EVP_PKEY_CTX *ctx, int type, int p1, void *p2) {
   }
 }
 
+static int pkey_dh_ctrl_str(EVP_PKEY_CTX *ctx, const char *type,
+                            const char *value) {
+  // We don't support:
+  // * dh_paramgen_prime_len
+  // * dh_rfc5114
+  // * dh_param
+  // * dh_paramgen_generator
+  // * dh_paramgen_subprime_len
+  // * dh_paramgen_type
+
+  if (strcmp(type, "dh_pad") == 0) {
+    char* str_end = NULL;
+    long pad = strtol(value, &str_end, 10);
+    if(str_end == value || pad < 0 || pad > INT_MAX) {
```

Comment:
is `pad == 0` valid here? looks like `EVP_PKEY_CTX_ctrl` [won't check](https://github.com/aws/aws-lc/blob/998227dd58fb44e141f0b2e6461209651db11e19/crypto/evp_extra/p_dh.c#L114-L137) this before setting it.

Reply:
My philosophy has been that the role of the `ctrl_str` function is to parse the provided string into a (roughly) acceptable value, then defer to the underlying function (e.g., `EVP_PKEY_CTX_set_dh_pad`) to deal with determine whether the specific value meets all pre-conditions. (I.e., I wanted to avoid redundancy in the checking of pre-conditions.)

Reply:
That makes sense. It looks like OpenSSL [allows this](https://github.com/openssl/openssl/blob/a57c6f84920bff522bca5fede73f1a3f132d7cff/crypto/dh/dh_pmeth.c#L266-L269) as well.